### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by fbclh**
**Project requirements:**

- Add redux-thunk to your project.
- Use fetch or axios for making API calls.
- Fetch book list from Bookstore API on Books page load.
- - Add an Action Creator, that returns a function.
- - Fetch list of books.
- - Dispatch another action (add an action type) that will pass API response to reducer and updates the state.
- Refactor your add book and remove book features to persist your changes in the server.
- -  Change your existing Action Creators, so they return functions.
- - Make changes by using API.
- - Dispatch an action (action type that existed before refactor) that will updates the state accordingly.
- Add book and remove book should work in the same way after the refactor.
- No styling is required.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -34,12 +18,49 @@ export const removeBook = (id) => ({
...
+};
+
+export const addBookToAPI = (payload) => async (dispatch) => {
+  dispatch(addBook(payload));
+  await fetch(URL, {
+    method: 'POST',
+    body: JSON.stringify(payload),
+    headers: { 'Content-type': 'application/JSON' },
+  });
+};
```
Comment (AnselemOdims) [inline src/redux/books/books.js:45]: - [ ] Please kindly ensure that the addition of the book persists. At the moment, when you add a new book and refresh then it disappears. That means your API request is not working properly. You can dispatch only after you are sure the fetch promise has been fulfilled using a `try-catch`  block.

File: src/redux/books/books.js [hunk 2]
```diff
@@ -34,12 +18,49 @@ export const removeBook = (id) => ({
...
+
+export const removeBookFromAPI = (id) => async (dispatch) => {
+  dispatch(removeBook(id));
+  await fetch(`${URL}/${id}`, {
+    method: 'DELETE',
+    body: JSON.stringify({ item_id: id }),
+    headers: { 'Content-type': 'application/JSON' },
+  });
+};
```
Comment (AnselemOdims) [inline src/redux/books/books.js:54]: - [ ] Please kindly also ensure that the removal of the book persists.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @fbclh, 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/redux/books/books.js (hunk 2 above)
Comment (fbclh) [reply src/redux/books/books.js:54]: Hi thanks for your feedback however on my end everything is working as expected, on refresh the book that was added persist and on remove click event the book is removed from the list.

**Review by AmaduKamara — APPROVED** [review]
# Status: Approved 🟢🟢🟢 

Hi @fbclh,

YesYouAreGIF

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**